src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -939,16 +1168,22 @@ private Type EmitFunctionCallNode(AssociativeNode node)
             EmitOpCode(OpCodes.Call, keygen);
 
             var local = DeclareLocal(typeof(IEnumerable<MethodBase>));
-
-            EmitOpCode(OpCodes.Ldloca, local);
+            //local could be null if we are not emitting currently.
+            if(local != null)
```

Comment:
Can you clarify under what circumstances can this be null?

Reply:
yes, during `GatherTypeInfo` - `DeclareLocal` will return null.

I think we should really remove all use of `ilGen` directly - and wrap all methods, I noticed it's being used in the new coercion methods, if you are okay with it I will wrap those calls in helpers so we can just return from them if the compiler phase is not emitIL?

essentially, it's not good to write anything to the IL stream until we are ready, since we can't remove it.